packages/sanes-chrome-extension/src/routes/account/index.tsx

```diff
@@ -5,60 +5,85 @@ import PageLayout from 'medulas-react-components/lib/components/PageLayout';
 import { ToastContext } from 'medulas-react-components/lib/context/ToastProvider';
 import { ToastVariant } from 'medulas-react-components/lib/context/ToastProvider/Toast';
 import Typography from 'medulas-react-components/lib/components/Typography';
-import { ACCOUNT_STATUS_ROUTE } from '../paths';
+import { ACCOUNT_STATUS_ROUTE, WELCOME_ROUTE } from '../paths';
 import SelectField, { Item } from 'medulas-react-components/lib/components/forms/SelectFieldForm';
 import { useForm } from 'react-final-form-hooks';
 import Form from 'medulas-react-components/lib/components/forms/Form';
 import { PersonaContext } from '../../context/PersonaProvider';
 import { AccountInfo } from '../../logic/persona/accountManager';
 import ListTxs from './components/ListTxs';
+import { history } from '../../store/reducers';
+import { sendGetPersonaMessage } from '../../extension/messages';
 
 const CREATE_NEW_ONE = 'Create a new one';
 
 const AccountView = (): JSX.Element => {
   const [accounts, setAccounts] = React.useState<Item[]>([]);
-  const persona = React.useContext(PersonaContext);
+  const personaProvider = React.useContext(PersonaContext);
   const { show } = React.useContext(ToastContext);
   const { form, handleSubmit } = useForm({
     onSubmit: () => {},
   });
 
   React.useEffect(() => {
     async function fetchMyAccounts(): Promise<void> {
-      const accounts = persona.accountNames;
+      const accounts = personaProvider.accountNames;
       let actualItems: Item[] = [];
       actualItems.push({ name: CREATE_NEW_ONE });
       accounts.forEach((acc: AccountInfo) => actualItems.push({ name: acc.name }));
       setAccounts(actualItems);
     }
 
     fetchMyAccounts();
-  }, [persona.accountNames]);
+  }, [personaProvider.accountNames]);
+
+  React.useEffect(() => {
+    async function fetchPersonaData(): Promise<void> {
+      const isExtensionContext = typeof chrome !== 'undefined';
+      if (!isExtensionContext) {
+        return;
+      }
+
+      const response = await sendGetPersonaMessage();
+      if (response) {
+        personaProvider.update(response.accounts, response.mnemonic, response.txs);
```

Comment:
Could we do this in `sanes-chrome-extension/src/index.tsx` already? Then we do not need 2 round trips of `sendGetPersonaMessage` and we don't need to handle the impossible case that `response` is falsy here.